src/reducers/resourceHelpers.js

```diff
@@ -181,27 +189,26 @@ export const updateValueErrors = (state, valueKey) => {
   const value = state.values[valueKey]
   const property = state.properties[value.propertyKey]
   const propertyTemplate = state.propertyTemplates[property.propertyTemplateKey]
-  // If this is first value, then must have a value.
-  const errors = []
-  if (
-    value.key === property.valueKeys[0] &&
-    propertyTemplate.required &&
-    propertyTemplate.type === "literal" &&
-    !value.literal
-  )
-    errors.push("Literal required")
+  // If this is first value, then errors must get a value.
+  let errors = []
+
+  if (propertyTemplate.type === "literal") {
+    errors.push(validateLiteralRequired(value, property, propertyTemplate))
+    if (value.literal !== null && value.literal !== "") {
+      errors.push(validateLiteralRegex(value, propertyTemplate))
+      errors.push(validateLiteralInteger(value, propertyTemplate))
+      errors.push(validateLiteralDateTime(value, propertyTemplate))
+      errors.push(validateLiteralDateTimeStamp(value, propertyTemplate))
+    }
+  }
 
   if (propertyTemplate.type === "uri") {
-    if (value.key === property.valueKeys[0] && propertyTemplate.required) {
-      if (!value.uri) errors.push("URI required")
-      if (!value.label) errors.push("Label required")
-    } else {
-      if (value.uri && !value.label) errors.push("Label required")
-      if (!value.uri && value.label) errors.push("URI required")
-    }
-    if (value.uri && !isValidURI(value.uri)) errors.push("Invalid URI")
+    errors.push(
+      validateUriPropertiesAndValue(value, property, propertyTemplate)
+    )
   }
 
+  errors = errors.flat(1) // all that pushing makes for nested arrays
```

Comment:
If the error functions above all return (one-dimensional) arrays, you could use array destructuring for this, I think? Especially since `errors` is already a mutable value. So, something like:

```javascript
let errors = []
// ...
errors = [
  ...errors, 
  ...literalRegexValidationError(value, propertyTemplate),
  ...literalIntegerValidationError(value, propertyTemplate),
  ...literalDateTimeValidationError(value, propertyTemplate),
  ...literalDateTimeStampValidationError(value, propertyTemplate),
]
// ... 
errors = [...errors, ...uriPropertiesAndValueErrors(value, property, propertyTemplate)]
```

You can see how it works in this dumbed-down example:

```javascript
> let errors = []
> const error1 = ['one']
> const error2 = ['two', 'three']
> const error3 = []
> const error4 = ['four']
> errors = [...errors, ...error1, ...error2, ...error3, ...error4]
[ 'one', 'two', 'three', 'four' ]
```